From bda3d4bc4cfe97c8a2f05299d1c58467ffb6674b Mon Sep 17 00:00:00 2001
From: Jaegeuk Kim <jaegeuk@kernel.org>
Date: Wed, 8 Jan 2020 15:10:02 -0800
Subject: [PATCH 256/420] f2fs: update f2fs document regarding to fsync_mode

This patch adds missing fsync_mode entry in f2fs document.

Fixes: 04485987f053 ("f2fs: introduce async IPU policy")
Reviewed-by: Chao Yu <yuchao0@huawei.com>
Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>
---
 Documentation/filesystems/f2fs.txt | 3 ++-
 fs/f2fs/segment.h                  | 5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/Documentation/filesystems/f2fs.txt b/Documentation/filesystems/f2fs.txt
index 00c112e6f459..2581fb269cfe 100644
--- a/Documentation/filesystems/f2fs.txt
+++ b/Documentation/filesystems/f2fs.txt
@@ -323,7 +323,8 @@ Files in /sys/fs/f2fs/<devname>
                               updates in f2fs. User can set:
                                0x01: F2FS_IPU_FORCE, 0x02: F2FS_IPU_SSR,
                                0x04: F2FS_IPU_UTIL,  0x08: F2FS_IPU_SSR_UTIL,
-                               0x10: F2FS_IPU_FSYNC, 0x40: F2FS_IPU_NOCACHE.
+                               0x10: F2FS_IPU_FSYNC, 0x20: F2FS_IPU_ASYNC,
+                               0x40: F2FS_IPU_NOCACHE.
                               Refer segment.h for details.
 
  min_ipu_util                 This parameter controls the threshold to trigger
diff --git a/fs/f2fs/segment.h b/fs/f2fs/segment.h
index 2473fdbdf9d9..99f1794c1d21 100644
--- a/fs/f2fs/segment.h
+++ b/fs/f2fs/segment.h
@@ -619,9 +619,10 @@ static inline int utilization(struct f2fs_sb_info *sbi)
  *                     threashold,
  * F2FS_IPU_FSYNC - activated in fsync path only for high performance flash
  *                     storages. IPU will be triggered only if the # of dirty
- *                     pages over min_fsync_blocks.
+ *                     pages over min_fsync_blocks. (=default option)
+ * F2FS_IPU_ASYNC - do IPU given by asynchronous write requests.
  * F2FS_IPU_NOCACHE - disable IPU bio cache.
- * F2FS_IPUT_DISABLE - disable IPU. (=default option)
+ * F2FS_IPUT_DISABLE - disable IPU. (=default option in LFS mode)
  */
 #define DEF_MIN_IPU_UTIL	70
 #define DEF_MIN_FSYNC_BLOCKS	8
-- 
2.15.0

